pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -0,0 +1,1089 @@
+	if err != nil {
+		return err
+	}
+	pdb.SetOwnerReferences(ownerRefs)
+	err = Mgr.GetClient().Update(context.TODO(), pdb)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (dst *PravegaCluster) updateControllerReferences(ownerRefs []metav1.OwnerReference) error {
+	configmap := &corev1.ConfigMap{}
+	name := dst.ConfigMapNameForController()
+	err := Mgr.GetClient().Get(context.TODO(),
+		types.NamespacedName{Name: name, Namespace: dst.Namespace}, configmap)
+	if err != nil {
+		return err
+	}
+	configmap.SetOwnerReferences(ownerRefs)
+	err = Mgr.GetClient().Update(context.TODO(), configmap)
+	if err != nil {
+		return err
+	}
+
+	svc := &corev1.Service{}
+	name = dst.ServiceNameForController()
+
+	err = Mgr.GetClient().Get(context.TODO(),
+		types.NamespacedName{Name: name, Namespace: dst.Namespace}, svc)
+	if err != nil {
+		return err
+	}
+	svc.SetOwnerReferences(ownerRefs)
+	err = Mgr.GetClient().Update(context.TODO(), svc)
+	if err != nil {
+		return err
+	}
+
+	pdb := &policyv1beta1.PodDisruptionBudget{}
+	name = dst.PdbNameForController()
+	err = Mgr.GetClient().Get(context.TODO(), types.NamespacedName{Name: name, Namespace: dst.Namespace}, pdb)
+	if err != nil {
+		return err
+	}
+	pdb.SetOwnerReferences(ownerRefs)
+	err = Mgr.GetClient().Update(context.TODO(), pdb)
+	if err != nil {
+		return err
+	}
+
+	deploy := &appsv1.Deployment{}
+	name = dst.DeploymentNameForController()
+	err = Mgr.GetClient().Get(context.TODO(), types.NamespacedName{Name: name, Namespace: dst.Namespace}, deploy)
+	if err != nil {
+		return fmt.Errorf("failed to get deployment (%s): %v", name, err)
+	}
+	deploy.SetOwnerReferences(ownerRefs)
+	err = Mgr.GetClient().Update(context.TODO(), deploy)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createConfigMap(p *v1alpha1.PravegaCluster) error {
+	cmName := "pravega-config"
+	cfgMap := &corev1.ConfigMap{}
+	err := Mgr.GetClient().Get(context.TODO(),
+		types.NamespacedName{Name: cmName, Namespace: p.Namespace}, cfgMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			cm := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ConfigMap",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cmName,
+					Namespace: p.Namespace,
+				},
+				Data: map[string]string{
+					"PRAVEGA_CLUSTER_NAME": p.Name,
+					"WAIT_FOR":             p.Spec.ZookeeperUri,
+				},
+			}
+			err = Mgr.GetClient().Create(context.TODO(), cm)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+func (p *PravegaCluster) migrateBookkeeper(srcObj *v1alpha1.PravegaCluster) error {
+	b := &bkapi.BookkeeperCluster{}
+	err := Mgr.GetClient().Get(context.TODO(), types.NamespacedName{Name: p.Name, Namespace: p.Namespace}, b)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			b.WithDefaults()
+			b.Name = p.GetName()
+			b.Namespace = p.GetNamespace()
+			err = createConfigMap(srcObj)
```

Comment:
why are we creating config map  here? This is created as part of bookkeeper-operator

Reply:
To use a bookkeeper cluster with Pravega, we need that config map.
Bk Operator documentation mentions it:
https://github.com/pravega/bookkeeper-operator/blob/master/doc/manual-installation.md#install-the-bookkeeper-cluster-manually